src/builtins/core/timezone.rs

```diff
@@ -121,6 +126,15 @@ impl UtcOffset {
     pub fn is_sub_minute(&self) -> bool {
         self.0 % NS_IN_MIN != 0
     }
+
+    /// Partial implementation of GetISODateTimeFor for a cached offset
+    pub(crate) fn get_iso_datetime_for(&self, instant: &Instant) -> TemporalResult<IsoDateTime> {
```

Comment:
issue (non-blocking): This should be infallible.

We're no longer fetching the offset anymore, so the primary getters should all be infallible. Looking at `from_epoch_nanos`. It doesn't need to be fallible either ... we should probably update that. It can be done in a separate PR though.

Reply:
That's a good point, yeah.